agent/agent-tooling/src/main/java/com/microsoft/applicationinsights/agent/internal/localstorage/LocalFileLoader.java

```diff
@@ -49,21 +53,24 @@ public LocalFileLoader(LocalFileCache localFileCache, File telemetryFolder) {
   // Load ByteBuffer from persisted files on disk in FIFO order.
   @Nullable
   ByteBuffer loadTelemetriesFromDisk() {
-    // TODO (heya) how does this load files from disk at startup that were left over from prior
-    //  process?
     String filenameToBeLoaded = localFileCache.poll();
-    if (filenameToBeLoaded == null) {
+    if (filenameToBeLoaded == null || filenameToBeLoaded.isEmpty()) {
       return null;
     }
 
     File tempFile;
+    File sourceFile;
```

Comment:
can you add a comment explaining why we are copying the file and then reading from the copy and then deleting the copy (instead of just reading from the source directly)?